### PR TITLE
Feature/1207/235 blob client code

### PR DESCRIPTION
**Pull request by algattik**
## What this PR changes/adds

Share blob client code in EDC and DPF

## Why it does that

End-to-end flow for blob storage transfers across EDC and DPF is not working (#231). One issue is that there are two classes named AzureBlobStoreSchema with different properties inside, so that different schemas are used for describing data transfers.

In addition, when using the emulator to test blob data transfer, it will be necessary to support connecting to two distinct storage accounts in the emulator. The existing implementation of `BlobStoreApiImplTest` only allowed overriding the storage endpoint to a fixed value, thus connecting to a single storage account.

## Further notes

Moved code previously in BlobAdapterFactory into BlobStoreApiImpl.

As part of follow-up PRs, we will provide further fixes including removing the use of the Shared Key in the data request, in favour of using Vault secrets.

## Linked Issue(s)

#235 

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: extensions/azure/blobstorage/blob-data-operator/build.gradle.kts [hunk 1]
```diff
@@ -20,8 +20,10 @@ val storageBlobVersion: String by project
 
 dependencies {
     api(project(":spi"))
+    api(project(":common:util"))
```
Comment (cpeeyush) [inline extensions/azure/blobstorage/blob-data-operator/build.gradle.kts:23]: nit: unused.

```suggestion
```

File: extensions/azure/data-plane/storage/build.gradle.kts [hunk 2]
```diff
@@ -12,32 +12,25 @@
...
 plugins {
     `java-library`
-    `java-test-fixtures`
 }
 
 
 dependencies {
-    implementation(project(":extensions:azure:data-plane:common"))
+    api(project(":extensions:data-plane:data-plane-spi"))
+    implementation(project(":extensions:azure:blobstorage:blob-core"))
```
Comment (cpeeyush) [inline extensions/azure/data-plane/storage/build.gradle.kts:25]: below it has `api(project(":extensions:azure:blobstorage:blob-core"))` as well ? Both needed ?

**Review by cpeeyush — APPROVED** [review]
Approved with suggestions.

**Review by marcgs — APPROVED** [review]

File: extensions/azure/blobstorage/blob-core/src/main/java/org/eclipse/dataspaceconnector/azure/blob/core/BlobStoreCoreExtension.java [hunk 3]
```diff
@@ -35,9 +35,10 @@ public String name() {
 
     @Override
     public void initialize(ServiceExtensionContext context) {
-        var blobstoreEndpoint = context.getSetting(EDC_BLOBSTORE_ENDPOINT, null);
+        var blobstoreEndpointTemplate = context
+                .getSetting(EDC_BLOBSTORE_ENDPOINT_TEMPLATE, "https://%s.blob.core.windows.net");
```
Comment (ouphi) [inline extensions/azure/blobstorage/blob-core/src/main/java/org/eclipse/dataspaceconnector/azure/blob/core/BlobStoreCoreExtension.java:39]: What is the value of using this "template" instead of the endpoint directly? Why did you need to do that?

Comment (algattik) [reply extensions/azure/blobstorage/blob-core/src/main/java/org/eclipse/dataspaceconnector/azure/blob/core/BlobStoreCoreExtension.java:39]: When the provider in system tests copies from source storage account to destination storage account in emulator, the endpoints are different (`"http://127.0.0.1:10000/" + accountName`)

File: extensions/azure/data-plane/storage/src/main/java/org/eclipse/dataspaceconnector/azure/dataplane/azurestorage/pipeline/AzureStorageDataSink.java [hunk 4]
```diff
@@ -33,7 +33,7 @@ public class AzureStorageDataSink extends ParallelSink {
     private String accountName;
     private String containerName;
     private String sharedKey;
-    private BlobAdapterFactory blobAdapterFactory;
+    private BlobStoreApi blobStoreApi;
```
Comment (ouphi) [inline extensions/azure/data-plane/storage/src/main/java/org/eclipse/dataspaceconnector/azure/dataplane/azurestorage/pipeline/AzureStorageDataSink.java:36]: I guess  you removed BlobAdapterFactory to simplify the code? A quick explanation about why you removed the BlobAdapterFactory and the changes it implies in the PR description would have helped for the review.

**Review by ouphi — APPROVED** [review]
